go/private/mode.bzl

```diff
@@ -125,6 +128,7 @@ def get_mode(ctx, host_only, go_toolchain, go_context_data):
         # You are not allowed to compile in race mode with pure enabled
         race = False
     debug = ctx.var["COMPILATION_MODE"] == "dbg"
+    linkobj = ctx.var.get("go.linkobj", "") == "1"
```

Comment:
I'm not sure why plugins don't work with `-linkobj`. To answer your question about what happens within Google, I don't believe plugins are supported at all; the Bazel rules have a greater scope and feature set than the internal rules.

If there's not a simple way to fix this, I'd rather not use `--define=go.linkobj=1` though. It would be better to just test `link == LINKMODE_PLUGIN` if that's the only affected mode.

Reply:
The reason it's not working is because: https://github.com/golang/go/issues/33820#issuecomment-524898207

I was using `link == LINKMODE_PLUGIN`, but it's not when building plugins, it's also when importing the `plugin` package, in regular go binaries... If any package imports `plugin`, then we can't use linkobj on any package.

So either we make it opt-in, or opt-out. There is an argument to be made about opt-out since `plugin` usage is so low, but still.

Reply:
Ah, interesting and unfortunate.

In general, I'm opposed to adding flags and additional configurations. More flags means testing is difficult, regressions are likely, and I can't afford to spend more time on support than I already do.

Maybe a workaround is possible for golang/go#33820 though? We could invoke the compiler without `-linkobj`. After it runs, the builder could copy the export data from the compiled .a file into the .x file. So the export data would appear in both archives. It's not as efficient as it could be with better toolchain support, but the I/O should be pretty cheap since the newly written .a file should still be in RAM.

Incidentally, I checked what Blaze does if a `go_binary` imports `"plugin"`: the linker just fails with the same error.